packages/react-hooks/src/hooks/steps.ts

```diff
@@ -0,0 +1,19 @@
+import {useState, useCallback} from 'react';
+
+export function useSteps(steps: number, initialStep = 0) {
+  const [currentStep, setStep] = useState(initialStep);
+
+  return {
+    currentStep,
+    nextStep: useCallback(() => {
+      if (currentStep < steps - 1) {
+        return setStep(currentStep + 1);
+      }
+    }, [currentStep, steps]),
+    previousStep: useCallback(() => {
+      if (currentStep > 0) {
+        return setStep(currentStep - 1);
```

Comment:
The return here is not doing much for you but is kind of confusing to read because it implies a value is being returned. I think this is clearer:

```suggestion
        setStep(currentStep + 1);
      }
    }, [currentStep, steps]),
    previousStep: useCallback(() => {
      if (currentStep > 0) {
        setStep(currentStep - 1);
```